modules/services/pantalaimon.nix

```
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.pantalaimon;
+
+  iniFmt = pkgs.formats.ini { };
+in {
+  meta.maintainers = [ maintainers.jojosch ];
+
+  options = {
+    services.pantalaimon = {
+      enable = mkEnableOption
+        "Pantalaimon - E2EE aware proxy daemon for matrix clients";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.pantalaimon;
+        defaultText = literalExample "pkgs.pantalaimon";
+        description = "The package to use for pantalaimon.";
+      };
+
+      settings = mkOption {
+        type = iniFmt.type;
+        default = { };
+        defaultText = literalExample "{ }";
+        example = literalExample ''
+          {
+            Default = {
+              LogLevel = "Debug";
+              SSL = true;
+            };
+            local-matrix = {
+              Homeserver = "https://matrix.org";
+              ListenAddress = "127.0.0.1";
+              ListenPort = 8008;
+            };
+          }
+        '';
+        description = ''
+          Configuration written to
+          <filename>~/.config/pantalaimon/pantalaimon.conf</filename>.
```

This is not necessarily true, since at least in theory XDG_CONFIG_HOME might be set differently. It's a bit of an odd piece of information in the first place, I would suggest to just drop it.

---

I've seen descriptions similar to this in a few places in HM, so I think this is fine. However, I would suggest changing this to reference `XDG_CONFIG_HOME`:

```suggestion
          <filename>$XDG_CONFIG_HOME/pantalaimon/pantalaimon.conf</filename>.
```